src/js/components/AppListItemComponent.jsx

```diff
@@ -47,43 +48,75 @@ var AppComponent = React.createClass({
   },
 
   onClick: function () {
-    this.context.router
-      .transitionTo("app", {appId: encodeURIComponent(this.props.model.id)});
+    var model = this.props.model;
+    var router = this.context.router;
+    if (model.isGroup) {
+      router.transitionTo("group", {groupId: encodeURIComponent(model.id)});
+    } else {
+      router.transitionTo("app", {appId: encodeURIComponent(model.id)});
+    }
+  },
+
+  getHealthBar: function () {
+    var model = this.props.model;
+    return (
+      <td className="text-right health-bar-column">
+        <AppHealthComponent model={model} />
+      </td>
+    );
+  },
+
+  getStatus: function () {
+    var model = this.props.model;
```

Comment:
Your are setting values on a single variable on different places, I want to avoid this.
This could easily become hard to maintain.